src/governance/classes.cpp

```diff
@@ -692,10 +711,48 @@ bool CSuperblock::IsExpired(const CGovernanceManager& governanceManager) const
     return false;
 }
 
-CGovernancePayment::CGovernancePayment(const CTxDestination& destIn, CAmount nAmountIn) :
+std::vector<uint256> CSuperblock::GetProposalHashes() const
+{
+    std::vector<uint256> res;
+
+    for (const auto& payment : vecPayments) {
+        res.push_back(payment.proposalHash);
+    }
+
+    return res;
+}
+
+std::string CSuperblock::GetHexStrData() const
+{
+    // {\"event_block_height\": 879720, \"payment_addresses\": \"yd5KMREs3GLMe6mTJYr3YrH1juwNwrFCfB\", \"payment_amounts\": \"5.00000000\", \"proposal_hashes\": \"485817fddbcab6c55c9a6856dabc8b19ed79548bda8c01712daebc9f74f287f4\", \"type\": 2}\u0000
+
+    std::string str_addresses = Join(vecPayments, "|", [&](const auto& payment) {
+        CTxDestination dest;
+        ExtractDestination(payment.script, dest);
+        return EncodeDestination(dest);
+    });
+    std::string str_amounts = Join(vecPayments, "|", [&](const auto& payment) {
+        return FormatMoney(payment.nAmount);
+    });
+    std::string str_hashes = Join(vecPayments, "|", [&](const auto& payment) { return payment.proposalHash.ToString(); });
+
+    std::stringstream ss;
+    ss << "{";
```

Comment:
why don't create here `UniValue`?

Such as:
```
UniValue obj(UniValue::VOBJ);
obj.pushKV("event_block_height", nBlockHeight);
...
return HexStr(obj.ToString())
```


btw, it looks like we use hash of this string, so, the string should be created by deterministic way, is it right? In this case indeed `UniValue` might be not suitable.

Reply:
Bcs changing version of library in future may change format of output, such as extra spaces in padding or extra \n at the end of file or maybe re-ordering objects in map

Reply:
Hmm yes, but I think the order should be always alphabetical. 
Anyway, I propose to keep current implementation for now.